### ISSUE TITLE
Creating custom operations for Tensorflow Windows

### ISSUE THREAD

**Issue by alayu** [tensorflow/tensorflow#30289]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Microsoft Windows 10
- ~~Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:~~
- TensorFlow installed from (source or binary): Source
- TensorFlow version: 1.12.3
- Python version: 3.5.0
- Installed using virtualenv? pip? conda?: conda
- Bazel version (if compiling from source): 0.17.2
- GCC/Compiler version (if compiling from source): 7.4.0
- ~~CUDA/cuDNN version: CPU, N/A~~
- ~~GPU model and memory: N/A~~

**Describe the problem**

I've been trying to get a custom operation working with a Windows version of TensorFlow. I've followed the guide:

https://www.tensorflow.org/guide/extend/op#build_a_pip_package_for_your_custom_op

I was able to build correctly, but it seems my custom operation is not recognized still when I try to run a test on it.

It seems to recognize the built-in custom operation tf.user_ops.my_fact, but it does not recognize the user_ops I created: squared_out.

**Provide the exact sequence of commands / steps that you executed before running into the problem**

1. Clone the repository:

```powershell
git clone https://github.com/tensorflow/tensorflow.git 
```

I put my C++ implementation of squared_out.cc in tensorflow\tensorflow\core\user_ops

```c++
#include "tensorflow/core/framework/op_kernel.h"
#include "tensorflow/core/framework/op.h"
#include "tensorflow/core/framework/shape_inference.h"

using namespace tensorflow;

REGISTER_OP("SquaredOut")
    .Input("to_square: int32")
    .Output("squared: int32")
    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext *c) {
        c->set_output(0, c->input(0));
        return Status::OK();
    });


class SquaredOutOp : public OpKernel {

    public:
        explicit SquaredOutOp(OpKernelConstruction* context) : OpKernel(context) {}

        void Compute(OpKernelContext* context) override {

            const Tensor& input_tensor = context->input(0);
            auto input = input_tensor.flat<int32>();

            Tensor* output_tensor = NULL;
            OP_REQUIRES_OK(context, context->allocate_output(0, input_tensor.shape(), &output_tensor));

            auto output_flat = output_tensor->flat<int32>();

            const int N = input.size();

            for(int i = 0; i < N; ++i){

                output_flat(i) = input(i) * input(i);

            }
        }
};

REGISTER_KERNEL_BUILDER(Name("SquaredOut").Device(DEVICE_CPU), SquaredOutOp);
```
2. Afterwards I configure and built the tensorflow library by using bazel:

```powershell
python ./configure.py
bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package
bazel-bin\tensorflow\tools\pip_package\build_pip_package C:/tmp/tensorflow_pkg
```

3. I then installed the .whl that was produced by bazel:

```powershell
pip install C:/tmp/tensorflow_pkg/tensorflow-version-cp35-cp35m-win_amd64.whl
```
4. Finally, I tried to test the newly imported squared_out from user_ops:
```python
import tensorflow as tf
#This works fine with build in fact function
tf.user_ops.fact
```
This returned:
```python
<function my_fact at 0x0000020CE5601048>
```
However, my custom operation doesn't work:
```python
tf.user_ops.squared_out
AttributeError: module 'tensorflow._api.v1.user_ops' has no attribute 'squared_out'
```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
Do I need to add a python binding for `squared_out` in tensorflow\tensorflow\python\user_ops? 

If so, I'm a little confused on how to do so.

**Comment by Scott-Thomas-Dev** [comment]
Hello alayu,

It would seem you missed a step, your operation is not automatically included in the build. You could add it to the build system if you're feeling adventurous or you can build it separately using bazel as explained in the tutorial.

specifically see: https://www.tensorflow.org/guide/extend/op#compile_the_op_using_bazel_tensorflow_source_installation

Furthermore once you load tensorflow you will need to load the op library in the python code 
so somewhere in your python code you'll need to add 
```
squared_out_lib = tf.load_op_library('./squared_out.so')
```

To summarize: 
You need to build the library then load it in python code.

**Comment by alayu** [comment]
@Scott-Thomas-Dev 

Thanks Scott, 

It seems like I was able to get the library compiled and loaded, I had an issue prior where .so file was not being read by Python. I believe I used Python 3.6.7 in a conda environment when compiling the .so file with bazel, but when I tried running it on my base environment the Python version was 3.5.0. 

The error I'm referring to was this:

```python
ImportError: DLL load failed: The specified module could not be found.
```

**Comment by BertramSun** [comment]
On Windows, I also have the same problem. Have you solved it?
